### PR TITLE
BNH-30 Add role for landlords

### PR DESCRIPTION
**Pull request by TomHanmer29**
added role in ClaimsTransformer for landlords to make landlord checks easier and some basic refactoring

### CODE REVIEW

File: web/Auth/ClaimsTransformer.cs [hunk 1]
```diff
@@ -45,6 +45,7 @@ public async Task<ClaimsPrincipal> TransformAsync(ClaimsPrincipal principal)
         var claims = new List<Claim>(existingClaimsIdentity.Claims);
         // If they're an admin in the database, add the role to their claims
         if (databaseUser.IsAdmin) claims.Add(new Claim(ClaimTypes.Role, "Admin"));
+        if (databaseUser.LandlordId!=null) claims.Add(new Claim(ClaimTypes.Role, "Landlord"));
```
Comment (emilypatsko) [inline web/Auth/ClaimsTransformer.cs:48]: spacing
```suggestion
        if (databaseUser.LandlordId != null) claims.Add(new Claim(ClaimTypes.Role, "Landlord"));
```

File: web/Auth/ClaimsTransformer.cs [hunk 2]
```diff
@@ -60,7 +61,7 @@ private async Task<UserDbModel> CreateUserInDatabase(string googleAccountId, Cla
             GoogleUserName = name,
             GoogleEmail = email,
             GoogleAccountId = googleAccountId,
-            IsAdmin = false
+            IsAdmin = false,
```
Comment (emilypatsko) [inline web/Auth/ClaimsTransformer.cs:64]: Not that it matters but any reason you added this comma? 😄

**Review by emilypatsko — APPROVED** [review]
One small formatting change but I'm otherwise happy, feel free to run it by @BenRamchandani too

File: web/Views/Shared/_Layout.cshtml [hunk 3]
```diff
@@ -23,17 +23,16 @@
                         <li class="nav-item">
                             <a class="nav-link text-dark" asp-area="" asp-controller="Home" asp-action="Index">Home</a>
                         </li>
-                        @if (User.Identity?.IsAuthenticated == true)
+                        @if (User.Identity?.IsAuthenticated == true && User.IsInRole("Admin"))
```
Comment (BenRamchandani) [inline web/Views/Shared/_Layout.cshtml:26]: Do we need the `User.Identity?.IsAuthenticated == true` check? I don't really mind keeping it for explicitness, but I would expect the `IsInRole` check to cover it.

**Review by BenRamchandani — APPROVED** [review]
Looks good, note you'll need to rebase to get the workflow file so the tests run.

File: web/Auth/ClaimsTransformer.cs (hunk 2 above)
Comment (TomHanmer29) [reply web/Auth/ClaimsTransformer.cs:64]: I added it when I was trying something else but forgot to remove it oops

File: web/Views/Shared/_Layout.cshtml (hunk 3 above)
Comment (TomHanmer29) [reply web/Views/Shared/_Layout.cshtml:26]: Yeah that's a very good point! I think I'll remove it for neatness :)

**Comment by TomHanmer29** [comment]
Going to close this and reopen it to see if the tests will work

**Pull request closed without merging**